doc/content/integrations/payload-formatters/javascript/uplink/_index.md

```diff
@@ -0,0 +1,256 @@
+---
+title: "Uplink"
+description: ""
+weight: 1
+aliases:
+  - "/integrations/payload-formatters/javascript/uplink-decoder"
+---
+
+The JavaScript `decodeUplink()` function is called when a data uplink message is received from a device. This function decodes the binary payload received from the end device to a human-readable JSON object that gets send upstream to the application. If present, the `normalizeUplink()` function is called to normalize the decoded payload in a common data structure.
+
+<!--more-->
+
+The decoder function takes an input object and returns an output object:
+
+```js
+function decodeUplink(input) {
+  // input has the following structure:
+  // {
+  //   "bytes": [1, 2, 3], // FRMPayload (byte array)
+  //   "fPort": 1
+  // }
+  return {
+    data: {
+      bytes: input.bytes
+    },
+    warnings: ["warning 1", "warning 2"], // optional
+    errors: ["error 1", "error 2"] // optional (if set, the message is dropped)
+  };
+}
+```
+
+The decoder function's output shown above is incorporated in the [`as.up.data.forward`]({{< ref "/reference/api/events#event:as.up.data.forward" >}}) event, where the `data.uplink_message` object contains:
+
+```json
+{
+  "f_port": 1,
+  "frm_payload": "AQID",
+  "decoded_payload": {
+    "bytes": [1, 2, 3]
+  }
+}
+```
+
+Uplink payload is transmitted as binary payload by the end device. {{% tts %}} shows this binary payload in the `frm_payload` field (Base64 encoded). The result of the `decodeUplink()` function is contained in the `decoded_payload` field.
+
+If an error is present in `errors`, the payload is invalid and the message will be dropped. Any warnings in `warnings` are informative.
```

Comment:
Actually we drop the decoded payload (so it is `nil`), but the uplink message itself will be forwarded. 
https://github.com/TheThingsNetwork/lorawan-stack/blob/a5445f0cf840e9587e4e918a3e50cd7b7b964136/pkg/applicationserver/payload.go#L151-L154